File: src/main/java/br/com/zumbolovsky/fateapp/domain/postgres/UserInfoRepository.java
package br.com.zumbolovsky.fateapp.domain.postgres;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserInfoRepository extends JpaRepository<UserInfo, Integer> {

    @Query("SELECT userInfo FROM UserInfo userInfo " +
            "JOIN FETCH userInfo.roles role " +
            "WHERE role.name = :role")
    List<UserInfo> findAllByRole(@Param("role") String role);

    @Query("SELECT userInfo FROM UserInfo userInfo " +
            "JOIN FETCH userInfo.roles role " +
            "WHERE userInfo.user = :user " +
            "AND userInfo.password = :password")
    Optional<UserInfo> findOneByUserAndPassword(
        @Param("user") String user,
        @Param("password") String password);

    @Query("SELECT userInfo FROM UserInfo userInfo " +
            "JOIN FETCH userInfo.roles role " +
            "WHERE userInfo.user = :user")
    Optional<UserInfo> findOneByUser(@Param("user") String user);
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/domain/postgres/UserInfoRepository.kt
package br.com.zumbolovsky.fateapp.domain.postgres

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserInfoRepository : JpaRepository<UserInfo, Int?> {

    @Query("""SELECT userInfo FROM UserInfo userInfo
        JOIN FETCH userInfo.roles role 
        WHERE role.name = :role""")
    fun findAllByRole(@Param("role") role: String): List<UserInfo>

    @Query("""SELECT userInfo FROM UserInfo userInfo
        JOIN FETCH userInfo.roles role 
        WHERE userInfo.user = :user 
        AND userInfo.password = :password""")
    fun findOneByUserAndPassword(
        @Param("user") user: String,
        @Param("password") password: String): Optional<UserInfo>

    @Query("""SELECT userInfo FROM UserInfo userInfo
        JOIN FETCH userInfo.roles role 
        WHERE userInfo.user = :user""")
    fun findOneByUser(@Param("user") user: String): Optional<UserInfo>
}